core/src/main/java/com/web3auth/core/Web3Auth.kt

```diff
@@ -165,6 +174,19 @@ class Web3Auth(web3AuthOptions: Web3AuthOptions) {
         //initiate keyStore
         initiateKeyStoreManager()
 
+        //fetch project config
+        fetchProjectConfig().whenComplete { response, error ->
+            if (error == null) {
+                web3AuthOption.originData = response.whitelist?.signed_urls
+                if (response?.whiteLabelData != null) {
+                    web3AuthOption.whiteLabel =
+                        web3AuthOption.whiteLabel?.merge(response.whiteLabelData)
+                }
+            } else {
+                initializeCf.completeExceptionally(Exception(Web3AuthError.getError(ErrorCode.SOMETHING_WENT_WRONG)))
+            }
```

Comment:
Also, whenCOmplete of fetchProjectConfig won't wait and call authorizeSession first. Please make it wait.
Use .get() or .join() instead